### PR TITLE
adding sparklines to records barchart

### PR DESCRIPTION
**Pull request by lindsayplatt**
- turned the records stacked bar chart into a function
- added file that creates the bar chart + sparklines
- added csv of records by characteristic type and year
- add png of the final sparklines/bar chart combo

### CODE REVIEW

**Comment by lindsayplatt** [comment]
@jread-usgs @ldecicco-USGS would love some eyes on the data cleaning I've done here. If you want to take a look at my ggplot stuff, I'd love some help! A lot of that was trial, error, and a little magic :sparkles:

**Comment by jordansread** [comment]
Loos so great @lindsaycarr!

File: R/carbon_analysis.R [hunk 1]
```diff
@@ -0,0 +1,72 @@
+## pull in data for carbon lakes analysis
```
Comment (lindsayplatt) [inline R/carbon_analysis.R:1]: Whoops, forgot my PR for the figure wasn't merged before pushing this. 

@ldecicco-USGS what do you think? Am I in the ballpark? Collated the groups of names using http://www.waterqualitydata.us/Codes/Characteristicname?mimeType=xml 

Need some help catching errors with the server 

``` r
> all_P_data <- lapply(stateCd$STUSAB[c(31, 40)], queryByState, characteristicNames = DIC_names)
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Server returned nothing (no headers, no data) 

# BUT works when I run them separately
all_DIC_list <- lapply(stateCd$STUSAB[31], queryByState, characteristicNames = DIC_names)
all_DIC_list2 <- lapply(stateCd$STUSAB[40], queryByState, characteristicNames = DIC_names)
```

Comment (ldecicco-USGS) [inline R/carbon_analysis.R:1]: If you can become an `lapply` expert, more power to you. That would be great. But for me....I've acknowledge my R shortcomings and reconciled that I'll never fully understand them, and therefore stick to old fashioned loops.

**Pull request merged**